### ISSUE TITLE
Handling Transfer-Encoding: chunked in injectResponse

### ISSUE THREAD

**Issue by confuciussayuhm** [gr3p1p3/transparent-proxy#36]
Hello there,

TL;DR: session.response.body doesn't play nice with `Transfer-Encoding: chunked`.

I have encountered something that I think the owners of this repository are more likely to address than a hack with Globals and injectResponse. When a server responds to a transparent-proxy request that is marked as a chunk using the `Transfer-Encoding: chunked` header in the response, transparent-proxy doesn't "group" all chunked responses for a particular request into one session.response.body object. Instead, transparent-proxy proxies each response with associated chunked bytes through to the client which then "groups" and render correctly (as the client understands what to do with `Transfer-Encoding: chunked`).

What I expect transparent-proxy should do, roughly, is to "break" the chunked stream by buffering each relevant chunked response then make available the session.response.body object in injectResponse once all chunked responses have arrived.

Or perhaps there is a way I can handle this `Transfer-Encoding: chunked` that is already built into transparent-proxy that I have missed?

**Comment by gr3p1p3** [comment]
Look at boolean `session.response.complete` like [this example](https://github.com/gr3p1p3/transparent-proxy/blob/master/examples/handleGzip.js)

**Comment by confuciussayuhm** [comment]
Thanks for the quick reply. I have updated my code to include this boolean `session.response.complete`. It seems that chunked responses never switch `session.response.complete` to `true`.

You can observe this. Use [this gist](https://gist.github.com/kadishmal/3871017) that will only respond in transfer-encoded chunks.

**Comment by gr3p1p3** [comment]
You are right, it was no handling for this.

Try with new version @1.12.5 and let me know :)

**Comment by confuciussayuhm** [comment]
Awesome. `session.response.complete` is now `true`. Following on from that, I now seem to be stuck decoding the `rawResponse` from the concatenated `rawResponseBodyChunks`. I'm getting this error from the `zlib.gunzip` line [in here](https://github.com/gr3p1p3/transparent-proxy/blob/master/examples/handleGzip.js): `Uncaught Error Error: Error: incorrect header check`. If I look at the two `rawResponseBodyChunks` in my case, I see they contain the same appended chunk bytes. I assume the relevant header bytes from the different chunks are not being stripped before being concatenated? And thus the zlib.gunzip step complains about data header?

**Comment by gr3p1p3** [comment]
How can I reproduce this error? It works fine for me
